File: src/java/tests/ai/nektron/grownet/tests/LateralBusDecayTests.java
package ai.nektron.grownet.tests;

import ai.nektron.grownet.LateralBus;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class LateralBusDecayTests {

    @Test
    public void busDecayResetsModulationAndIncrementsStep() {
        LateralBus bus = new LateralBus();
        bus.setInhibitionFactor(1.0);
        bus.setModulationFactor(2.7);
        long before = bus.getCurrentStep();

        bus.decay();

        assertEquals(0.9, bus.getInhibitionFactor(), 1e-12);
        assertEquals(1.0, bus.getModulationFactor(), 0.0);
        assertEquals(before + 1, bus.getCurrentStep());
    }
}

File: src/cpp/tests/LateralBusDecayTests.cpp
// Simple test for LateralBus decay semantics
#ifdef GTEST_AVAILABLE
#include <gtest/gtest.h>
#endif

#include "LateralBus.h"

#ifdef GTEST_AVAILABLE
TEST(LateralBusDecay, ResetsModulationAndIncrementsStep) {
    grownet::LateralBus bus;
    bus.setInhibitionFactor(1.0);
    bus.setModulationFactor(2.5);
    auto before = bus.getCurrentStep();
    bus.decay();
    EXPECT_NEAR(bus.getInhibitionFactor(), 0.9, 1e-12);
    EXPECT_DOUBLE_EQ(bus.getModulationFactor(), 1.0);
    EXPECT_EQ(bus.getCurrentStep(), before + 1);
}
#endif

